lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -696,9 +709,171 @@ migrateManually tr proxy defaultFieldValues =
     addScriptAddressGapIfMissing :: Sqlite.Connection -> IO ()
     addScriptAddressGapIfMissing conn =
         addColumn_ conn True (DBField SeqStateScriptGap) value
-      where
+     where
         value = T.pack $ show $ Seq.getAddressPoolGap Seq.defaultAddressPoolGap
 
+    -- This migration is rather delicate. Indeed, we need to introduce an
+    -- explicit 'fee' on known transactions, so only do we need to add the new
+    -- column (easy), but we also need to find the right value for that new
+    -- column (delicate).
+    --
+    -- Note that it is not possible to recover explicit fees on incoming
+    -- transactions without having access to the entire ledger (we do not know
+    -- the _amount_ from inputs of incoming transactions). Therefore, by
+    -- convention it has been decided that incoming transactions will have fee
+    -- equals to 0.
```

Comment:
No no. `0` is the value we want here for it conveys the right intuition: if I didn't make that transaction, from my perspective fees are 0. Since wallets are already user-centric (which is the reason why we can't know the amount any arbitrary inputs, but only those belonging to the wallet), it's a reasonable justification.